### PR TITLE
Overhaul DEVELOPMENT

### PR DESCRIPTION
**Pull request by allentiak**
TODO: make Release Checklist use current release workflow

Thanks to FastChat!

https://github.com/lm-sys/FastChat
https://lmarena.ai/

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -104,26 +132,19 @@ Such components reside in a namespace. There exists a constructor function calle
...
+* `halt-key!`` implements `AutoCloseable` to ensure resources are properly released when the node is closed.
+* `m/pre-init-spec` provides a spec for the dependency map to ensure correct configuration.
+* `ig/init-key` initializes the node and logs a meaningful message at info level.
+* `ig/halt-key!` closes the node and releases any held resources.
 
-It is important to avoid using reflection. In order to see reflection warnings, make sure to use ```(set! *warn-on-reflection* true)``` in every namespace which does Java interop.
+### Function Specs
 
-## REPL
+Every public function should have a spec. Function specs are declared in a namespace with the suffix `_spec` appended to the function's namespace. Public module-level function specs reside in the `src` folder, while inner-module public function specs reside in the `test` folder. This ensures that specs are used in tests but not included in the global classpath, reducing the uberjar and memory footprint.
```
Comment (alexanderkiel) [inline DEVELOPMENT.md:143]: _spec -> -spec

**Review by alexanderkiel — CHANGES_REQUESTED** [review]

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -1,82 +1,110 @@
...
 
 ## Style Guide
 
-The Clojure code in this project mainly follows the [Clojure Style Guide][2], enforced by `cljfmt`. For more details, please check `cljfmt.edn`."
-
-## Pure Functions
+Follow the [Clojure Style Guide][2], enforced by `cljfmt`. For more details, check the `cljfmt.edn` file.
 
-For most parts Blaze is implemented using pure functions. Pure function depend only on their arguments and only produce an output without doing any side effects. Pure functions have one important property, they are referentially transparent, meaning it doesn't matter when or how often they are called.
+### Pure Functions
```
[the remark this answers is not in the document]
Comment (allentiak) [reply DEVELOPMENT.md:87]: Yeap.
After all, they all discuss coding conventions.

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -1,82 +1,110 @@
...
+* nodejs
+* GNU Make
+* [Clojure, with CLI tools](https://clojure.org/guides/install_clojure)
+* cljfmt: `clj -Ttools install io.github.weavejester/cljfmt '{:git/tag "`[`<latest-stable-release>`](https://github.com/weavejester/cljfmt/releases/latest)`"}' :as cljfmt`
+
+#### Steps
+
+1. Run `make -C job-ig build` to create FHIR profiles.
+2. Run `make uberjar` to create the uberjar in the `target` directory.
+3. Run `docker build -t blaze:latest .` to build the Docker image.
```
[the remark this answers is not in the document]
Comment (allentiak) [reply DEVELOPMENT.md:29]: @alexanderkiel 
If you point me to the frontend steps, I could add them as see how the document looks afterwards.

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -1,82 +1,110 @@
...
+* code coverage (which should only increase on each commit).
+For more details, see the  `.github/` directory.
+
+### Configuration
 
 The configuration of the development system is done with the same environment variables used in the production system.
 Documentation: [Environment Variables](docs/deployment/environment-variables.md).
 
-## Release Checklist
+## Release Checklist (TODO: use current release workflow)
```
[the remark this answers is not in the document]
Comment (allentiak) [reply DEVELOPMENT.md:71]: I mean the workflow you currently use to create the releases.
The workflow currently detailed here corresponds to the `git-flow`-based workflow you used to follow.

File: DEVELOPMENT.md [hunk 5]
```diff
@@ -1,82 +1,110 @@
-# Development
+# Development Guide for Blaze
 
 ## Building Blaze
 
-The main build artefact of Blaze is a single Docker image. Apart from the Docker image, an uberjar is build, which can be used also.
+Blaze is built as a single Docker image, along with an uberjar for standalone use.
```
[the remark this answers is not in the document]
Comment (allentiak) [reply DEVELOPMENT.md:5]: Could you please point me to it?
I cannot find it either on https://github.com/samply/blaze/releases nor on https://hub.docker.com/r/samply/blaze/tags...

File: DEVELOPMENT.md [hunk 6]
```diff
@@ -1,82 +1,110 @@
...
-* merge the release branch back into develop
-* create release notes on GitHub
+1. Create a release branch named `release-v<version>`, e.g., `release-v0.29.0`.
+2. Update all occurrences of the old version (e.g., `0.28.0`) to the new version (e.g., `0.29.0`).
+3. Update the CHANGELOG based on the milestone.
+4. Create a commit with the title `Release v<version>`.
+5. Create a PR from the release branch to master.
+6. Merge the PR.
+7. Create and push a tag named `v<version>`, e.g., `v0.13.1`, on master at the merge commit.
+8. Merge the release branch back into develop.
```
[the remark this answers is not in the document]
Comment (allentiak) [reply DEVELOPMENT.md:80]: You mean step 8, right?

File: DEVELOPMENT.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (alexanderkiel) [reply DEVELOPMENT.md:5]: It is a separate image: https://hub.docker.com/r/samply/blaze-frontend/tags

File: DEVELOPMENT.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (alexanderkiel) [reply DEVELOPMENT.md:30]: It's `make build-frontend`

File: DEVELOPMENT.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (alexanderkiel) [reply DEVELOPMENT.md]: yes

File: DEVELOPMENT.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexanderkiel) [reply DEVELOPMENT.md:87]: Ok Code Conventions but not Style Guide. So I would see an h2 called Code Conventions here with a h3 follow up called Style Guide.

**Review by alexanderkiel — CHANGES_REQUESTED** [review]

File: DEVELOPMENT.md [hunk 7]
```diff
@@ -1,82 +1,126 @@
...
+
+```make uberjar```
+
+3. Build the Blaze Docker image:
+
+```docker build -t blaze:latest .```
+
+4. Build the frontend Docker image:
+
+```make frontend```
```
Comment (alexanderkiel) [inline DEVELOPMENT.md:44]: It's `make build-frontend`

**Review by alexanderkiel — CHANGES_REQUESTED** [review]

**Review by alexanderkiel — APPROVED** [review]

**Pull request merged**